Widget Command Livewire Path Solution

ksaif534 opened the pull request:
Hello.

This is a solution to this Github issue: [https://github.com/filamentphp/filament/issues/14637](https://github.com/filamentphp/filament/pull/url)

### **Description**

When the make:filament-widget command is run to create a new widget and the resource is provided and when the user selects [App\Livewire] alongside other Livewire components after selecting Table from the first prompt, a file permission error was thrown and the directory was not being created. The solution now creates a new file TestWidget.php within the app/Livewire directory when it's prompted by the user.

### **Visual Changes**

**After the Solution:**

<img width="751" alt="widget-command-livewire-solution" src="https://github.com/user-attachments/assets/d30b719c-472c-4db8-960d-78be76a1760c" />

**Before the Solution:**

<img width="1012" alt="widget-command-livewire-solution-before" src="https://github.com/user-attachments/assets/4be4e7bb-37a4-4ea4-ae1b-f9d54daec004" />

### **Functional Changes**

I've fixed the style with the composer cs command, the file changes have been tested.

ksaif534 committed:
Widget Command Livewire Solution

ksaif534 committed:
Style Fixes for PR (Widget Command Livewire Solution)

fathurrohman26 commented:
Hi @kasfi, I have tried the solution you provided, but it seems the generated namespace is not correct.

> ###
> <img alt="@ksaif534" width="24" height="24" src="https://avatars.githubusercontent.com/u/28676259?s=48&amp;v=4"> ** [ksaif534](/ksaif534)**

```bash
❯ php artisan make:filament-widget PaymentChart --chart --resource=PaymentResource

 ┌ Where would you like to create this? ────────────────────────┐
 │ [App\Livewire] alongside other Livewire components           │
 └──────────────────────────────────────────────────────────────┘

 ┌ Which type of chart would you like to create? ───────────────┐
 │ Bar chart                                                    │
 └──────────────────────────────────────────────────────────────┘

   INFO  Filament widget [app/Livewire/PaymentChart.php] created successfully.

   INFO  Make sure to register the widget in `PaymentResource::getWidgets()`, and then again in `getHeaderWidgets()` or `getFooterWidgets()` of any `PaymentResource` page.
```
```bash

❯ bat app/Livewire/PaymentChart.php
───────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: app/Livewire/PaymentChart.php
───────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ <?php
   2   │
   3   │ namespace \PaymentResource\Widgets;
   4   │
   5   │ use Filament\Widgets\ChartWidget;
   6   │
   7   │ class PaymentChart extends ChartWidget
   8   │ {
   9   │     protected static ?string $heading = 'Chart';
  10   │
  11   │     protected function getData(): array
  12   │     {
  13   │         return [
  14   │             //
  15   │         ];
  16   │     }
  17   │
  18   │     protected function getType(): string
  19   │     {
  20   │         return 'bar';
  21   │     }
  22   │ }
───────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

ksaif534 commented:
@fathurrohman26 Oh, I didn't notice that. I'll try to fix it.

danharrin committed:
clean up

danharrin commented:
I did remove the test since there is a new style of tests in v4 and this would not work being merged in to that

ksaif534 committed:
Merge branch 'filamentphp:3.x' into 3.x

ksaif534 committed:
chore: fix code style

ksaif534 committed:
Resolve merge conflict for WidgetCommandLivewireTest.php

ksaif534 committed:
chore: fix code style

ksaif534 commented:
@fathurrohman26 @danharrin I've adjusted the changes and fixed the problems in the initial solution. It'd be great if you can provide feedback.

Adjusted PR: https://github.com/filamentphp/filament/pull/15910
